### ISSUE TITLE
Feature request: add bibliography from DOI

### ISSUE THREAD

**Issue by es132** [texstudio-org/texstudio#745]
I believe it would be great to be able to add a bibliography entry just by copy/pasting a DOI. It is possible to download a BibTeX entry with a single web request as explained in https://citation.crosscite.org/docs.html in sections 4 and 5.

Example use: I am reading the PDF of a paper and I want to cite it. I copy the DOI from the PDF. In TeXstudio I click on the Bibliography menu, then on an "Add from DOI" entry, a dialog appears with a box to paste the DOI, the bibtex entry is downloaded according to the doc above and it is inserted at the cursor position or at the end of the bib file.

**Comment by muzimuzhi** [comment]
This sounds like a function that a typical Reference Management Software will provide. And, by searching, I found there are already some implementations of the doi2bib function.
 - website https://www.doi2bib.org/
 - webpage https://scipython.com/apps/doi2bib/, based on the simple python script described on https://scipython.com/blog/doi-to-bibtex/

**Comment by es132** [comment]
> This sounds like a function that a typical Reference Management Software will provide.

That's right. I do understand that you do not want to implement a reference management software and I agree with your decision. However sometimes or for some people, a reference management software is overkill.

>     * webpage https://scipython.com/apps/doi2bib/, based on the simple [python script](https://scipython.com/blog/doi-to-bibtex/) and requesting bib info from http://dx.doi.org/

This is the kind of approach I had in mind. Note that these days, https://doi.org/ is preferred.

**Comment by es132** [comment]
I implemented this feature as a macro. I share it here in case other people are interested.
It was written for Linux. For Windows, replace the `cmd='wget ...` line with:
`cmd='powershell iwr -outf %TMP%/doi2bib.tmp https://api.crossref.org/works/'+doi[i]+'/transform/application/x-bibtex;cat %TMP%/doi2bib.tmp > /dev/null';"`

To use it, copy/paste one or more DOI in the editor, select them and run the macro.

```
%SCRIPT
var text=new String(),cmd=new String(),out=new String();
var proc,doi;
var re=new RegExp('10.[0-9]+/\\S+',"g");
text=cursor.selectedText();
doi = text.match(re);
for (var i=0;i<doi.length;i++) {
  cmd='wget --quiet --output-document=- https://api.crossref.org/works/'+doi[i]+'/transform/application/x-bibtex > /dev/null';
  proc=system(cmd,"");
  proc.waitForFinished();
  if (proc.exitCode()==0) {
    out+=proc.readAllStandardOutputStr()+"\n";
  }
  else {
    out+="Couldn't retrieve "+doi[i]+"\n";
  }
}
cursor.replaceSelectedText(out);
delete(text,cmd,out,proc,doi,re);
```
